sonic-sensormond/scripts/sensormond

```diff
@@ -0,0 +1,559 @@
+                                            )
+            warning = warning | current_status.over_threshold
+
+            if current != NOT_AVAILABLE and current_status.set_under_threshold(current, low_threshold):
+                self._log_on_status_changed(not current_status.under_threshold,
+                                            'Low current warning cleared: {} current restored to {}mA, low threshold {}mA'.
+                                            format(name, current, low_threshold),
+                                            'Low current warning: {} current current {}mA, low threshold {}mA'.
+                                            format(name, current, low_threshold)
+                                            )
+            warning = warning | current_status.under_threshold
+
+            fvs = swsscommon.FieldValuePairs(
+                [('current', str(current)),
+                ('minimum_current', str(minimum_current)),
+                ('maximum_current', str(maximum_current)),
+                ('high_threshold', str(high_threshold)),
+                ('low_threshold', str(low_threshold)),
+                ('warning_status', str(warning)),
+                ('critical_high_threshold', str(high_critical_threshold)),
+                ('critical_low_threshold', str(low_critical_threshold)),
+                ('is_replaceable', str(is_replaceable)),
+                ('timestamp', datetime.now().strftime('%Y%m%d %H:%M:%S'))
+                ])
+
+            self.table.set(name, fvs)
+            if self.is_chassis_system and self.chassis_table is not None:
+                self.chassis_table.set(name, fvs)
+        except Exception as e:
+            self.log_warning('Failed to update isensor status for {} - {}'.format(name, repr(e)))
+
+    def _remove_isensor_from_db(self, isensor, parent_name, isensor_index):
+        name = try_get(isensor.get_name, '{} isensor {}'.format(parent_name, isensor_index + 1))
+        self.table._del(name)
+
+        if self.chassis_table is not None:
+            self.chassis_table._del(name)
+
+#
+# Daemon =======================================================================
+#
+class SensorMonitorDaemon(daemon_base.DaemonBase):
+
+    # Initial update interval
+    INITIAL_INTERVAL = 5
+    # Periodic Update interval 
+    UPDATE_INTERVAL = 60
+    # Update time threshold. If update time exceeds this threshold, log warning msg.
+    UPDATE_ELAPSED_THRESHOLD = 30
+
+    def __init__(self):
+        """
+        Initializer of SensorMonitorDaemon
+        """
+        super(SensorMonitorDaemon, self).__init__(SYSLOG_IDENTIFIER)
+
+        # Set minimum logging level to INFO
+        self.set_min_log_priority_info()
+
+        self.stop_event = threading.Event()
+
+        self.wait_time = self.INITIAL_INTERVAL
+
+        self.interval  = self.UPDATE_INTERVAL
+
+        self.chassis = sonic_platform.platform.Platform().get_chassis()
+
```

Comment:
We might want to add try/except block to check if chassis is not None and exit if it is instead of handling it going forward

Reply:
I added a try except for getting chassis and added error log. However I am not sure an exit is a good idea at this point as the process will continuously exit on restart. Since this is an optional feature at this point, I think an error syslog should be good enough.